upup/pkg/fi/fitasks/keypair.go

```diff
@@ -180,15 +185,22 @@ func (_ *Keypair) Render(c *fi.Context, a, e, changes *Keypair) error {
 			createCertificate = true
 		} else if changes.Type != "" {
 			createCertificate = true
+		} else if a.KeypairType == "" {
+			// The a.KeysetAPIVersion will == "" when the keyset is a legacy keyset and does not have a keypair.yaml
+			// file store in the state store
+			createCertificate = true
```

Comment:
Why are you setting createCertificate here?

Reply:
I can test without it, but I think my testing did not create the keypair.yaml.

Reply:
As we discussed, pretty sure this shouldn't drive creation of a new certificate, but should instead drive a new `UpdateStoredVersion` or similar call